sympy/tensor/tensor.py

```diff
@@ -1771,7 +1774,20 @@ def __call__(self, *indices, **kw_args):
         A(a, -b)
 
         """
-        tensor = Tensor(self, indices, **kw_args)
+
+        updated_indices = []
+        for idx, typ in zip(indices, self.index_types):
+            if isinstance(idx, str):
+                if idx.startswith('-'):
+                    updated_indices.append(-TensorIndex(idx[1:], typ))
```

Comment:
This feels like a slippery slope - why allow `A('-a')` but not `A('b-a')` etc...

Reply:
Indeed, it is theoretically possible to implement more ways to write the indices. I don't have any strong opinion on it, so I leave it to other people to discuss how useful it is. I did not go further because this would complicate a bit more the parsing and it's not as useful: adding string arguments help to make the code much less verbose, but going further that way is a marginal improvement.

Reply:
I'm almost arguing the reverse - performing any string parsing at all here feels odd to me.

Reply:
Sure, I would have preferred to avoid it. But, without this line, there is no way to write a covariant index in string format since the unary `-` does not work on strings.

Reply:
Ah, apologies - I did not understand what `-` did here. _maybe_ clearer as:

```suggestion
                if idx.startswith('-'):
                    updated_indices.append(TensorIndex(idx[1:], typ, is_up=False))
```